### PR TITLE
Feature/theme

### PR DESCRIPTION
**Pull request by RameshDPrasad**
Update theme. The theme colours are not set for all possible options as design reference is not available;able.

### CODE REVIEW

File: Sources/DolbyIOUIKit/Theme/Asset/ColorAttributes.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
+//
+//  Colors.swift
```
Comment (CasperMok) [inline Sources/DolbyIOUIKit/Theme/Asset/ColorAttributes.swift:2]: `ColorAttributes.swift`

File: Sources/DolbyIOUIKit/Theme/Asset/ColorAttributes.swift [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+public struct ToggleAttribute {
+    let textColor: Color
+    let tintColor: Color
+    let focusedBackgroundColor: Color
+    let outlineColor: Color
+}
+
+// MARK: IconButton Colors
+
+public struct IconAttribute {
```
Comment (CasperMok) [inline Sources/DolbyIOUIKit/Theme/Asset/ColorAttributes.swift:54]: Shall it be `IconButtonAttribute`? or shall the comment be updated?

File: Sources/DolbyIOUIKit/Theme/Asset/IconAsset.swift [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 import Foundation
 
-public enum ImageAsset: String, ImageAssetable {
+public enum IconAsset: String {
```
Comment (CasperMok) [inline Sources/DolbyIOUIKit/Theme/Asset/IconAsset.swift:7]: Update the corresponding filename in the comment

File: Sources/DolbyIOUIKit/Theme/Asset/Theme.swift [hunk 4]
```diff
@@ -3,38 +3,158 @@
...
-    associatedtype _ColorAsset: ColorAssetable
-    associatedtype _FontAsset: FontAssetable
-    associatedtype _ImageAsset: ImageAssetable
-
-    subscript(colorAsset: _ColorAsset) -> Color? { get }
-    subscript(fontAsset: _FontAsset) -> Font { get }
-    subscript(imageAsset: _ImageAsset) -> Image { get }
-}
+public protocol Theme {
+    // Material theme color definitions
```
Comment (CasperMok) [inline Sources/DolbyIOUIKit/Theme/Asset/Theme.swift:7]: Not sure if we shall mention the Material theme here. It is Android. Open to this.

File: Sources/DolbyIOUIKit/Theme/Asset/ThemeManager.swift [hunk 5]
```diff
@@ -7,12 +7,12 @@ import Foundation
 import SwiftUI
 
 public class ThemeManager: ObservableObject {
-    static public let shared: ThemeManager = ThemeManager()
+    static public let instance: ThemeManager = ThemeManager(theme: DefaultTheme())
```
Comment (CasperMok) [inline Sources/DolbyIOUIKit/Theme/Asset/ThemeManager.swift:10]: I think `shared` or `sharedInstance` are more commonly used naming in iOS. @aravind-raveendran Any comment?

File: Sources/DolbyIOUIKit/Theme/Asset/Theme.swift (hunk 4 above)
Comment (RameshDPrasad) [reply Sources/DolbyIOUIKit/Theme/Asset/Theme.swift:7]: It is not specific to Android, other platforms also use it

File: Sources/DolbyIOUIKit/Theme/Asset/ThemeManager.swift (hunk 5 above)
Comment (aravind-raveendran) [reply Sources/DolbyIOUIKit/Theme/Asset/ThemeManager.swift:10]: `shared` is the commonly used name.

File: Sources/DolbyIOUIKit/Theme/Asset/Theme.swift [hunk 6]
```diff
@@ -3,38 +3,158 @@
...
-public typealias Theme = AbstractTheme<ColorAsset, FontAsset, ImageAsset>
+public enum ButtonStyles {
+    case primary
+    case primaryDanger
+    case secondary
+    case secondaryDanger
+    case tertiary
+}
+
+public enum TextStyles {
```
[the remark this answers is not in the document]
Comment (RameshDPrasad) [reply Sources/DolbyIOUIKit/Theme/Asset/Theme.swift:144]: No our design system definitions are bit different (based on M2 version of material design whereas I have tried to use M3), these styles are derived from material design (M3) and yes they should specify the font and size as well but right now it doesn't, I will move it eventually after reconciling with our design system, right now it is all over the place

[the remark this answers is not in the document]
Comment (aravind-raveendran) [reply Sources/DolbyIOUIKit/Theme/Asset/Theme.swift:144]: Taking a style `displayLarge` as example - what attributes of a `Text` component do they control and what would be the values?

File: Sources/DolbyIOUIKit/Theme/Asset/Theme.swift (hunk 4 above)
Comment (aravind-raveendran) [reply Sources/DolbyIOUIKit/Theme/Asset/Theme.swift:7]: Looking at its implementation in `DefaultTheme` they are all set to use `UIColor.neonPurple400` at the moment. Is that as expected?

File: Sources/DolbyIOUIKit/Theme/Asset/Theme.swift (hunk 6 above)
[the remark this answers is not in the document]
Comment (RameshDPrasad) [reply Sources/DolbyIOUIKit/Theme/Asset/Theme.swift:144]: Font, size, weight etc. https://m3.material.io/styles/typography/type-scale-tokens

[the remark this answers is not in the document]
Comment (RameshDPrasad) [reply Sources/DolbyIOUIKit/Theme/Asset/Theme.swift:144]: For color definition, eventually the color swatches would go away and just the material color definitions would remain which will map to the components properties and hence the components attributes would not be required to be exposed in the theme.

File: Sources/DolbyIOUIKit/Theme/Asset/Theme.swift (hunk 4 above)
Comment (RameshDPrasad) [reply Sources/DolbyIOUIKit/Theme/Asset/Theme.swift:7]: Will update it in future, set it to default `UIColor.neonPurple40` as we don't have all the definitions in the design yet

File: Sources/DolbyIOUIKit/Theme/Asset/Theme.swift (hunk 6 above)
[the remark this answers is not in the document]
Comment (aravind-raveendran) [reply Sources/DolbyIOUIKit/Theme/Asset/Theme.swift:144]: Cool 👍

**Review by aravind-raveendran — APPROVED** [review]

**Pull request merged**